simplepso/pso.py

```diff
@@ -378,7 +360,7 @@ def run_ssa(self, model, num_particles, num_iterations, num_sim,
             pass
         else:
             self._setup_pso()
-        if self.simulator is None:
+        if simulator is None:
             raise Exception("Must provide an SSA simulator to use this method")
```

Comment:
`ValueError` would be more specific here?

Reply:
Done!